src/drivers/display/drm/lv_linux_drm_egl.c

```diff
@@ -0,0 +1,715 @@
+/**
+ * @file lv_linux_drm_egl.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_linux_drm.h"
+
+#if LV_USE_LINUX_DRM && LV_LINUX_DRM_USE_EGL
+
+#include <fcntl.h>
+#include <string.h>
+#include <xf86drmMode.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <gbm.h>
+#include <drm_fourcc.h>
+#include <xf86drm.h>
+#include <time.h>
+#include <unistd.h>
+#include "lv_linux_drm_egl_private.h"
+#include "../../../draw/lv_draw_buf.h"
+#include "../../opengles/lv_opengles_debug.h"
+
+#include "../../opengles/lv_opengles_driver.h"
+#include "../../opengles/lv_opengles_texture.h"
+#include "../../opengles/lv_opengles_private.h"
+
+#include "../../../stdlib/lv_string.h"
+#include "../../../display/lv_display.h"
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+typedef struct {
+    int fd;
+    struct gbm_bo * bo;
+    uint32_t fb_id;
+} drm_fb_state_t;
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static uint32_t tick_cb(void);
+static void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_map);
+static void event_cb(lv_event_t * e);
+
+static lv_result_t drm_device_init(lv_drm_ctx_t * ctx, const char * path);
+static void drm_device_deinit(lv_drm_ctx_t * ctx);
+
+static lv_egl_interface_t drm_get_egl_interface(lv_drm_ctx_t * ctx);
+static drmModeConnector * drm_get_connector(lv_drm_ctx_t * ctx);
+static drmModeModeInfo * drm_get_mode(lv_drm_ctx_t * ctx);
+static drmModeEncoder * drm_get_encoder(lv_drm_ctx_t * ctx);
+static drmModeCrtc * drm_get_crtc(lv_drm_ctx_t * ctx);
+static void drm_on_page_flip(int fd, unsigned int frame, unsigned int sec, unsigned int usec, void * data);
+static drm_fb_state_t * drm_fb_state_create(lv_drm_ctx_t * ctx, struct gbm_bo * bo);
+static void drm_fb_state_destroy_cb(struct gbm_bo * bo, void * data);
+static void drm_flip_cb(void * driver_data, bool vsync);
+
+static void * drm_create_window(void * driver_data, const lv_egl_native_window_properties_t * properties);
+static void drm_destroy_window(void * driver_data, void * native_window);
+static size_t drm_egl_select_config_cb(void * driver_data, const lv_egl_config_t * configs, size_t config_count);
+static inline void set_viewport(lv_display_t * display);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+lv_display_t * lv_linux_drm_create(void)
+{
+    lv_tick_set_cb(tick_cb);
+    lv_drm_ctx_t * ctx = lv_zalloc(sizeof(*ctx));
+    LV_ASSERT_MALLOC(ctx);
+    if(!ctx) {
+        LV_LOG_ERROR("Failed to create drm context");
+        return NULL;
+    }
+
+    ctx->display = lv_display_create(1, 1);
```

Comment:
The first display created will decide padding sizes of the default theme. Using 1x1 will choose small padding permanently, even after the true display resolution is set to e.g. 1920x1080.

See #7385

Best idea for now is to match what lv_linux_drm.c does.

```suggestion
    ctx->display = lv_display_create(800, 480);
```